File: LeetCode/Medium/0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        int n = 1 ; 
        ListNode* tail = head ;
        if(head == nullptr || head-> next == nullptr || k == 0 ){
            return head;
        }
        while(tail->next != nullptr){
            tail = tail->next;
            n++;
        }
        tail -> next = head ;
        int total_cuts = n - (k%n);
        ListNode* newtail = head;
        for(int i = 1 ; i < total_cuts ; i++){
            newtail = newtail->next;
        }
        ListNode* Newhead = newtail->next;
        newtail->next = nullptr ;
        return Newhead;


    }
};

File: LeetCode/Medium/0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if(  head == None or  head.next == None or k == 0):
            return head
        c = 1 
        tail = head
        while tail.next:
            tail = tail.next
            c += 1
        tail.next = head
        effective_cut = c - (k%c)
        new_tail = head
        for i in range(1,effective_cut):
            new_tail = new_tail.next
        new_head = new_tail.next
        new_tail.next = None
        return new_head

        